src/Tempest/Database/DatabaseDriverFactory.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Database;
+
+use Tempest\Database\Drivers\MySqlDriver;
+use Tempest\Database\Drivers\PostgreSqlDriver;
+use Tempest\Database\Drivers\SQLiteDriver;
+
+final class DatabaseDriverFactory
+{
+    public static function make(DatabaseDialect $dialect, array $options): DatabaseDriver
```

Comment:
was thinking of adding a default to the `$options` parameter but argued against it as we want the user to pass options